### PR TITLE
[SE-4304] feat: add upload folder prefix

### PR DESCRIPTION
**Pull request by gabor-boros**
Implement additional folder structure options. This way, users could introduce as many custom subfolders into the S3 folder structure as they need to automate their processes.

**Dependencies**: None

**Screenshots**:  N/A

**Sandbox URL**: https://studio.stage.campus.gov.il/admin/

**Merge deadline**: ASAP

**Testing instructions**:

* Enable task https://courses.stage.campus.gov.il/admin/djcelery/periodictask/4/
* Wait for the time period (about 1 minute)
* Go to https://courses.stage.campus.gov.il/admin/instructor_task/instructortask/?o=-6
* Check for the latest 5 runs for:
  * ccx-v1:OpenCraft+CS101+2020_T1+ccx@29
  * ccx-v1:OpenCraft+CS101+2020_T1+ccx@28
  * ccx-v1:OpenCraft+CS101+2020_T1+ccx@23
  * ccx-v1:OpenCraft+CS101+2020_T1+ccx@19
  * course-v1:OpenCraft+CS101+2020_T1
* Check for the related reports within the SE-4304 folder on S3 in the [corresponding grades bucket](https://s3.console.aws.amazon.com/s3/buckets/stage-olivex-student-grades?prefix=stage%2FSE-4304%2F&region=eu-west-1)
* Disable the periodic task https://courses.stage.campus.gov.il/admin/djcelery/periodictask/4/

**Author notes and concerns**:

N/A

**Reviewers**

- [ ] @itsjeyd 
- [ ] @pkulkark

### CODE REVIEW

File: lms/djangoapps/periodic_instructor_reports/tasks.py [hunk 1]
```diff
@@ -85,7 +86,14 @@ def periodic_task_wrapper(course_ids, *args, **kwargs):
 
         if use_folders_by_date:
             task_kwargs.update({
-                "upload_parent_dir": date.today().strftime("%Y/%m/%d")
+                "upload_parent_dir": "{prefix}{folder}".format(
+                    prefix=upload_folder_prefix,
+                    folder=date.today().strftime("%Y/%m/%d")
+                )
```
Comment (itsjeyd) [inline lms/djangoapps/periodic_instructor_reports/tasks.py:92]: @gabor-boros What happens if `upload_folder_prefix` is not defined?

```
>>> "{foo}{bar}".format(foo=None, bar="baz")
'Nonebaz'
```

**Review by itsjeyd — COMMENTED** [review]
@gabor-boros I have a couple questions about these changes (aside from the in-line comment above):

* The test task (https://courses.stage.campus.gov.il/admin/djcelery/periodictask/4/) has `include_ccx` set to `true`. But after enabling the test task the [S3 bucket](https://s3.console.aws.amazon.com/s3/buckets/stage-olivex-student-grades?region=eu-west-1&prefix=stage/se-4304/2021/05/26/&showversions=false) only received files for the main course.
* The number of instructor tasks that were created for the first run of the periodic task was as expected (5 tasks for 1 main course and 4 CCX courses). But on subsequent runs multiple instructor tasks were generated per course (cf. https://courses.stage.campus.gov.il/admin/instructor_task/instructortask/?o=-6).

Could you please have a look?

File: lms/djangoapps/periodic_instructor_reports/tasks.py (hunk 1 above)
Comment (gabor-boros) [reply lms/djangoapps/periodic_instructor_reports/tasks.py:92]: As upload_folder_prefix set to `""` it would not matter.

**Comment by gabor-boros** [comment]
> @gabor-boros I have a couple questions about these changes (aside from the in-line comment above):
> 
>     * The test task (https://courses.stage.campus.gov.il/admin/djcelery/periodictask/4/) has `include_ccx` set to `true`. But after enabling the test task the [S3 bucket](https://s3.console.aws.amazon.com/s3/buckets/stage-olivex-student-grades?region=eu-west-1&prefix=stage/se-4304/2021/05/26/&showversions=false) only received files for the main course.
> 
>     * The number of instructor tasks that were created for the first run of the periodic task was as expected (5 tasks for 1 main course and 4 CCX courses). But on subsequent runs multiple instructor tasks were generated per course (cf. https://courses.stage.campus.gov.il/admin/instructor_task/instructortask/?o=-6).
> 
> 
> Could you please have a look?

I did investigate the issue. It was happening because one of the stage instances had a wrong version of the configuration branches, therefore the `ENABLE_COURSE_FILENAME_CCX_SUFFIX` was disabled. Hence, the CCX files were saved with the exact same name as non-ccx files and overwrote each other.

**Comment by clemente** [comment]
:+1:

- [X] I tested this: as mentioned in instructions: enabled, then it ran for a few minutes and created 5 reports each time (parent course, and 1 for each CCX), as expected. The files were successfully created in the S3 folders in the right paths (the prefix chosen in the task configuration)
- [X] I read through the code, very simple
- [X] I checked for accessibility issues
- [X] Includes documentation

File: lms/djangoapps/periodic_instructor_reports/tasks.py (hunk 1 above)
Comment (itsjeyd) [reply lms/djangoapps/periodic_instructor_reports/tasks.py:92]: @gabor-boros Ah right, for some reason that didn't stick around in my short-term memory when I reviewed this code 🙃 Thanks for clarifying.

**Pull request merged**